### ISSUE TITLE
model.train() returns None metrics after subprocess multi-GPU training

### ISSUE THREAD

**Issue by AffanBinFaisal** [ultralytics/ultralytics#24500]
## Description
When training with multiple GPUs from Python without \LOCAL_RANK\ in the environment, the parent process launches workers via subprocess and never runs \_do_train()\. The parent \BaseTrainer\ therefore keeps \alidator=None\, while \Model.train()\ only did \self.metrics = getattr(self.trainer.validator, 'metrics', None)\, which always resolves to \None\.

## Expected
\YOLO(...).train(device=[0,1], ...)\ (or \device='0,1'\) should return the same kind of validation metrics object as single-GPU training when training completes successfully.

## Actual
\self.metrics\ is \None\ on the parent process after subprocess DDP training.

## Context
- \BaseTrainer.validator\ is created in \_setup_train()\ (\get_validator()\), which only runs inside \_do_train()\.
- Subprocess DDP path: \BaseTrainer.train()\ runs \subprocess.run(generate_ddp_command(...))\ and returns without calling \_do_train()\ on the parent.

## Proposed fix
Run a single \al()\ on the loaded best/last checkpoint in \Model.train()\ when \	rainer.validator is None\ (with \plots=False\ and a single-GPU \device\ derived from multi-GPU training args when needed).

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AffanBinFaisal, thank you for the clear bug report and proposed fix for Ultralytics 🚀 This is an automated response to help move things forward quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

Since this appears to be a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if possible. A small standalone script showing the multi-GPU `model.train()` call, along with your package version, environment details, and any relevant logs would help us reproduce and verify the `None` metrics behavior more efficiently 🔍

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

The proposed fix is too heavy compared to the benefit. If the user needs the metrics, they could run validation themselves. We don't want to force a separate val run because most users do not use the returned metrics
